File: 2023/day06/part1.py
def read_file():
  with open('input.txt', encoding="utf-8") as f:
    return f.read()
  
def main():
  input = read_file()
  rows = [row.split(":")[1] for row in input.split("\n")]
  times = [int(num) for num in list(filter(bool, rows[0].split(" ")))]
  distances = [int(num) for num in list(filter(bool, rows[1].split(" ")))]

  race_wins = 0

  # races
  for i in range(len(times)):
    time = times[i]
    distance = distances[i]

    speed, wins = 0, 0
    while speed <= time:
      travel_dist = speed * (time - speed)

      if travel_dist > distance:
        wins += 1

      speed += 1

    if race_wins == 0:
      race_wins = wins
    else:
      race_wins *= wins

    print('race', i + 1, 'time', time, 'distance', distance, 'wins', wins)
    
  print(race_wins)

if __name__ == '__main__':
  main()

File: 2021/day09/part1.js
const fs = require("fs").promises;

async function main() {
  const input = await fs.readFile("input.txt", "utf8");
  const matrix = input
    .split("\n")
    .filter(Boolean)
    .map((r) => r.split("").map((x) => parseInt(x)));

  const lowPoints = [];

  for (let i = 0; i < matrix.length; i++) {
    for (let j = 0; j < matrix[i].length; j++) {
      if (i - 1 >= 0) if (matrix[i - 1][j] <= matrix[i][j]) continue;
      if (j - 1 >= 0) if (matrix[i][j - 1] <= matrix[i][j]) continue;
      if (i + 1 < matrix.length) if (matrix[i + 1][j] <= matrix[i][j]) continue;
      if (j + 1 < matrix[i].length)
        if (matrix[i][j + 1] <= matrix[i][j]) continue;

      lowPoints.push(matrix[i][j]);
    }
  }

  console.log(lowPoints.reduce((acc, val) => acc + val + 1, 0));
}

main();
